src/test/java/teammates/test/cases/webapi/DeleteCourseActionTest.java

```diff
@@ -30,87 +28,40 @@ protected String getRequestMethod() {
 
     @Override
     @Test
-    protected void testExecute() throws Exception {
+    public void testExecute() throws Exception {
+
         InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
         String instructorId = instructor1OfCourse1.googleId;
+        String courseId = instructor1OfCourse1.courseId;
 
-        loginAsInstructor(instructorId);
-
-        ______TS("Not enough parameters");
-
-        verifyHttpParameterFailure();
-
-        ______TS("Typical case, 2 courses");
+        ______TS("Typical case, delete a soft-deleted course in Recycle Bin");
```

Comment:
Add a case where a course is not in recycle bin? -> it should fail

Reply:
It doesn't fail?

Reply:
If it just `execute()`, then it won't. But with access control `checkSpecificAccessControl`, it will. Try put the test in `testAccessControl`?